imap_processing/hi/l1a/housekeeping.py

```diff
@@ -0,0 +1,79 @@
+"""Unpack IMAP-Hi housekeeping data."""
+import collections
+import dataclasses
+
+import xarray as xr
+
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import calc_start_time
+from imap_processing.hi.hi_cdf_attrs import (
+    hi_hk_l1a_attrs,
+    hi_hk_l1a_metadata_attrs,
+)
+
+
+def create_dataset(packets):
+    """Create dataset for each metadata field.
+
+    Parameters
+    ----------
+    packets : list
+        packet list
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    metadata_arrays = collections.defaultdict(list)
+    description_dict = {}
+
+    for data_packet in packets:
+        # Add metadata to array
+        for key, value in (data_packet.header | data_packet.data).items():
+            # convert key to lower case to match SPDF requirement
+            data_key = key.lower()
+            metadata_arrays[data_key].append(value.raw_value)
+            # description should be same for all packets
+            description_dict[data_key] = (
+                value.long_description or value.short_description
+            )
+
+    epoch_converted_time = [
+        calc_start_time(sc_time) for sc_time in metadata_arrays["ccsds_met"]
+    ]
```

Comment:
Not for this PR, but it might be nice if `calc_start_time` could handle an array and you could instead call:
`epoch_converted_times = calc_start_time(metadata_arrays["ccsds_met"])`

Reply:
I agree. Many of us are doing this. Would be nice to call a common util function for this.